### ISSUE TITLE
Oil Lamp Cooker Does Not Reload

### ISSUE THREAD

**Issue by Ichthysdekilt** [CleverRaven/Cataclysm-DDA#41933]
# Describe the bug
When attempting to reload oil lamp cookers it consumes the oil but does not fill.

# Steps To Reproduce

1. Acquire an oil lamp cooker
2. Create lamp oil (I made from tainted fat, I don't know if the source is relevant) and store in a bucket
3. Reload oil lamp cooker
4. Observe oil depletes but the cooker does not charge

# Expected behavior

Cooker charges/reloads/fills with oil, rather than transporting it into the shadow realm.

# Screenshots

# Versions and configuration

 - OS: W10
 - Game Version: 10815
 - Ingame language: English
 - Mods loaded: "dda",  "no_npc_food",  "secronom",  "aftershock",  "blazeindustries",  "crt_expansion",  "magiclysm",  DinoMod",  "alt_map_key",  "crazy_cataclysm",  "MMA",  "zets_hair_extensions_undead",  "Graphical_Overmap",  "Graphical_Overmap_Magiclysm",  "fuji_mpp",  "speedydex",  "stats_through_kills",  "StatsThroughSkills"

# Additional context

**Comment by MatiasFito** [comment]
This looks interesting, i will take a look and let you know what i can find. Most likely the magazine in the lamp oil cooker is incompatible with the lamp oil, but not enough that it won't let you try to recharge it, but incompatible enough to not hold it and take it to shadow realm as you said lol.

**Comment by MatiasFito** [comment]
I've found the issue, as i suspected the lamp oil cooker magazine accepted lamp oil, but it was not watertight so the lamp oil being in "phase: liquid" could not be held by the container. I took the opportunity to audit the cooking tools since a few also have liquid ammo, like gasoline, and also fixed those that were not watertight, and rigid (because they are made of steel, so the volume should not change based on content). I'll link to closing this issue in the PR.
